x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.test.ts

```diff
@@ -0,0 +1,296 @@
+        mount({
+          dataLoading: true,
+        });
+        RelevanceTuningLogic.actions.onInitializeRelevanceTuning(relevanceTuningProps);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchSettings,
+          schema,
+          dataLoading: false,
+          schemaConflicts,
+        });
+      });
+    });
+
+    describe('setSearchSettings', () => {
+      it('should set state', () => {
+        mount({
+          unsavedChanges: false,
+        });
+        RelevanceTuningLogic.actions.setSearchSettings(searchSettings);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchSettings,
+          unsavedChanges: true,
+        });
+      });
+    });
+
+    describe('setFilterValue', () => {
+      it('should set state', () => {
+        mount();
+        RelevanceTuningLogic.actions.setFilterValue('foo');
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          filterInputValue: 'foo',
+        });
+      });
+    });
+
+    describe('setSearchQuery', () => {
+      it('should set state', () => {
+        mount();
+        RelevanceTuningLogic.actions.setSearchQuery('foo');
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          query: 'foo',
+        });
+      });
+    });
+
+    describe('setSearchResults', () => {
+      it('should set state', () => {
+        mount({
+          resultsLoading: true,
+        });
+        RelevanceTuningLogic.actions.setSearchResults(searchResults);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchResults,
+          resultsLoading: false,
+        });
+      });
+    });
+
+    describe('setResultsLoading', () => {
+      it('should set state', () => {
+        mount({
+          resultsLoading: false,
+        });
+        RelevanceTuningLogic.actions.setResultsLoading(true);
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          resultsLoading: true,
+        });
+      });
+    });
+
+    describe('clearSearchResults', () => {
+      it('should set state', () => {
+        mount({
+          searchResults: [{}],
+        });
+        RelevanceTuningLogic.actions.clearSearchResults();
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchResults: null,
+        });
+      });
+    });
+
+    describe('resetSearchSettingsState', () => {
+      it('should set state', () => {
+        mount({
+          dataLoading: false,
+        });
+        RelevanceTuningLogic.actions.resetSearchSettingsState();
+
+        expect(RelevanceTuningLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: true,
+        });
+      });
+    });
+
+    describe('dismissSchemaConflictCallout', () => {
+      it('should set state', () => {
```

Comment:
I've gone back and forth on this, & I know `should set state` is shorter and easier to copy, but it would personally help me to get a more verbose/human-readable preview of what all reducers the action affects. For example for this one:

```js
    describe('dismissSchemaConflictCallout', () => {
      it('should set showSchemaConflictCallout to false', () => {
```

for the one above:

```js
    describe('resetSearchSettingsState', () => {
      it('should set dataLoading to true', () => {
```

And especially for the tests where multiple reducers are set and we're grabbing abstracted vars, it's just easier to immediately grok the intention/goal:

```js
    describe('onInitializeRelevanceTuning', () => {
      it('should set searchSettings, schema, & schemaConflicts from the API, and set dataLoading to false', () => {
```

which is a little verbose sure, but I tend to (as a dev) err on the side of providing more context via comments etc. than fewer 🤷

Reply:
Oh sorry I should have prefaced, this is a super optional comment and mostly me just thinking out loud, happy to be dissuaded otherwise if others feel more strongly about this

Reply:
I like more descriptive descriptions, I think they'd make better individual tests though, as descriptions get super long.

Reply:
021aea4

Reply:
Haha yeah it's definitely a tradeoff of which you prefer seeing more/less of. I think I slightly prefer longer test names to more it() block cruft (because I am a king of run of sentences :trollface: ), but it's like not really a strong feeling and I'm good both ways

Reply:
Yeah, I think as long as we're intentional about making our tests human understandable, we'll be fine. Separate blocks may make sense in some places, descriptions in others. These descriptions are 100% better than "should set state". And after I looked at trying to split it out into separate blocks I was like gross.